coediting/hooks/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependency
+ */
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependency
+ */
+import { BlockEdit, getBlockDefaultClassname, getBlockType, hasBlockSupport } from '@wordpress/blocks';
+import { addFilter } from '@wordpress/hooks';
+import { getWrapperDisplayName } from '@wordpress/element';
+
+/**
+ * Internal dependency
+ */
+import './style.scss';
+// TODO: Move selectors to the local folder
+import {
+	getBlock,
+	getFrozenBlockCollaboratorColor,
+	getFrozenBlockCollaboratorName,
+	isBlockFrozenByCollaborator,
+} from '../../editor/selectors';
+
+const withFrozenMode = ( OriginalComponent, originalProps ) => {
+	const Component = ( { isFrozenByCollaborator, ...props } ) => {
+		if ( isFrozenByCollaborator ) {
+			const { block, collaboratorColor, collaboratorName } = props;
+			const { attributes, name, isValid, uid } = block;
+			const blockType = getBlockType( name );
+
+			// Determine whether the block has props to apply to the wrapper.
```

Comment:
@aduth @youknowriad I copied it from the original component. Isn't something that should be part of the `block` object? If I follow properly the flow is as follows:
- in a completely different place use registered block type to create `block`
- her fetch block and its `attributes`
- get block type again to generate `wrapperProps` based on block's `attributes`

Reply:
>Isn't `wrapperProps ` something that should be part of the `block` object? If I follow properly the flow is as follows:

I'd rather see this move to a block supports. Originally I'd intended for the block list to be unaware of alignment features, but I think it's such a base functionality that it's fine. Might be even better implemented as a hook.